### DISCUSSION TITLE
Quais livros ler para melhorar seu código?

### DISCUSSION THREAD

**Discussion by Raffesmind** [frontendbr/forum#55] [Discussão]
Pessoas lindas desse Brasil baronil, quais são os livros que vocês recomendam ler para melhorar sua capacidade de desenvolvimento? Tanto especifico para o front, quanto bons para engenharia, quanto para design.

Qual seria o nível ou momento recomendado para a leitura? 
Penso que seria bem bacana fazer um livro com os livros recomendados pela galera, como no exemplo do JSPath.

**Comment by karlprieb** [comment]
Para JS começaria com [Eloquent Javascript](http://eloquentjavascript.net/).
Muito bom :)

**Comment by robsonbittencourt** [comment]
Clean Code do Uncle Bob

**Comment by vitorbritto** [comment]
@Raffesmind livros para otimizar sua capacidade em programar ou aguçar seu raciocínio lógico, não vejo como sendo um livro de uma linguagem de programação especifica. Tem que ser voltado para uma característica que precise / necessite ser melhorada (a sua forma de criar algoritmos, neste caso).

Eu diria que o _Clean Code_, mencionado pelo @robsonbittencourt, é um bom livro para nortear suas habilidades em criar algoritmos reusáveis, modularizados, manuteníveis e transparentes.

Além deste, indico o _Code Simplicity_, _Becoming a Better Programmer_ e _The Art of Readable Code_.

IMHO, para aprimorar suas habilidades em uma linguagem de programação, você precisa ler código e praticar.

Bons estudos! =]

**Comment by tcelestino** [comment]
Tem a versão [pt-br](http://braziljs.github.io/eloquente-javascript/) do Eloquent Javascript.

Uma dica que me deram tempos atrás é ver códigos de projetos open-source, você consegue aprender bastante.

**Comment by robisson** [comment]
Um que recomendo que é muito é o Javascript Application Design do Nicolás Bevacqua . É um livro em inglês mas o conteúdo é excelente. E nada que um Google translate não possa ajudar. Nele o autor demonstra várias técnicas dos workflows que estão sendo muito utilizados hoje e também melhores práticas para se programar  e ter um código que possa ter vida longa para uma manutenção decente.

**Comment by vitorbritto** [comment]
Concordo com o @tcelestino. Se aprende muito lendo código de projetos open-source.

**Comment by victormiguez** [comment]
Acho bem legal essas dicas, principalmente isso de ler códigos open-source.
Eu sempre trabalhei com front-end, focado em HTML e CSS mesmo, de um ano pra cá tenho me interessado mais por JavaScript, hoje tenho um nível que considero bom, mas às vezes sinto dificuldade em desenvolver um algoritmo mais complexo, e que eu consiga desenvolver com um código que me deixe realmente satisfeito.
Vocês acham que estudar uma linguagem back-end auxilia nesse ponto?

**Comment by robisson** [comment]
Eu não diria @victormiguez  que estudar uma linguagem backend via necessariamente lhe dar mais experiências em algorítimos complexos, há necessidades no desenvolvimento frontend que dependendo do que se está desenvolvendo, necessitam de algoritmos tão complexos quanto um código de backend.

Concordo com o @tcelestino  que ler outros códigos ajuda e ajuda muito, porque você pode ver como outras pessoas resolveram um problema que talvez você fizesse por outra abordagem. Eu particularmente sempre fui muito curioso em como os frameworks funcionam, isso porque a maioria dos frameworks largamente utilizados geralmente utilizam as melhores práticas de desenvolvimento, acho que são bons cases para se estudar.  Acho legal analisar os códigos do [](http://todomvc.com/). Que mostra como fazer a mesma coisa com diversos frameworks frontend e sem framework algum também..

Eu leio também bastante, se quiser umas dicas de livros para algoritmos e boas práticas javascript recomendo esses dois:

**Data Structures and Algorithms with Javascript de Michael Mcmillan** 
**Maintainable JavaScript Nicholas Zakas**

**Comment by hugooliveirad** [comment]
O livro que mudou a minha carreira foi o **Functional JavaScript** do Michael Fogus.

http://shop.oreilly.com/product/mobile/0636920028857.do

Apesar de concordar que melhorar o código vai muito além do que um livro de JavaScript. 

Pra mim a vantagem de ler um sobre JavaScript foi que eu conseguia aplicar diariamente as coisas que lia.

**Comment by douglas-castilho** [comment]
e falando em livros, um link legal (não vi se já postaram por aqui)
http://jsbooks.revolunet.com/

**Comment by doug2k1** [comment]
Andam recomendando muito a série "You Don't Know JS". Dá pra ler online de graça:
https://github.com/getify/You-Dont-Know-JS

**Comment by heliomsolivas** [comment]
Não sei se foi aqui que achei essa lista, mas estou seguindo a lista de livros, 
https://github.com/javascript-society/javascript-path

**Comment by vitor-mendrone-deel** [comment]
Acabei de ler o [CSS Secrets](http://shop.oreilly.com/product/0636920031123.do) da Lea Verou e recomendo fortemente, é um livro sensacional!

**Comment by eduardojmatos** [comment]
@Mendrone foi tema do último Meetup CSS esse livro. É realmente sensacional :)
